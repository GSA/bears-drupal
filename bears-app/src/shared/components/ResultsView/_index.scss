#usagov-bears-app {
  .result-view-details {
    padding: 24px;

    .step-back-link {
      color: #979797;
    }
  }

  .result-view-benefits,
  .result-view-unmet,
  .result-view-relvant-benefits,
  .result-view-share-results {
    padding: 16px 0;
    border-bottom: 2px solid lightgrey;
  }

  .result-view-unmet {
    button {
      margin-bottom: 16px;
    }

    p {
      padding-bottom: 16px;
    }
  }

  .result-view-share-results {
    text-align: center;
  }

  .result-view-share-results-button-group {
    button {
      margin-bottom: 8px;
    }
  }

  @media (width >= 1049px) {
    .result-view-details {
      padding: 24px 164px;
    }
  }
}
