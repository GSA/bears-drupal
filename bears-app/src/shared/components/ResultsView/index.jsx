import PropTypes from 'prop-types'
import { BenefitAccordionGroup, Heading, StepBackLink } from '../index'

/**
 * a functional component that renders a view of the form benefit state values
 * @component
 * @param {function} handlStepBack inherited ui translations
 * @param {object} ui inherited ui translations
 * @param {array} data inherited benefits data
 * @param {func} setBenefitsArray inherited state handler
 * @param {array} stepDataArray inherited state of inupt values from form entry
 * @return {html} returns a view page of filtered benefits
 */
const ResultsView = ({
  handleStepBack,
  ui,
  data,
  // setBenefitsArray,
  // stepDataArray,
}) => {
  // console.log(data)
  const { heading, stepBackLink } = ui

  // collect all the criteria keys and selected criteria values into an array
  // const selectedCriteria = stepDataArray
  //   .map((item, i) =>
  //     item.section.fieldsets.map(item => {
  //       return {
  //         id: item.fieldset.inputs[0].inputCriteria.id,
  //         values: item.fieldset.inputs[0].inputCriteria.values.find(
  //           value => value.selected
  //         ),
  //       }
  //     })
  //   )
  //   .flat() // we flatten all to have only one array

  // Total Criteria = y
  // Met Criteria = x
  // Not Met Criteria = z

  // Criteria Met Length	Label
  // x === y	"Likely Eligible"
  // z === 0 && x === undefined length > 0 "More Information Needed"
  // Criteria Not Met	Label
  // z > 0	"Not Eligible"

  // collect "Likely Eligible"

  // for each benefit eligiblity compare compare selectedCriteria
  // if there is a criteriakey match in a benefit
  // check that the value === acceptable values

  // function likelyEligible(selectedCriteria, data) {
  //   console.log(selectedCriteria)
  //   data.forEach((item, i) => {
  //     console.log(item.benefit.eligibility)
  //   })
  // }

  // console.log(likelyEligible(selectedCriteria, data))

  // compare the selected criteria array with benefits
  return (
    <div>
      <Heading headingLevel={2}>{heading}</Heading>
      {/* map all the benefits into cards */}
      <BenefitAccordionGroup data={data} entryKey={'benefit'} expandAll />
      {/* {data.map((item, i) => {
        return console.log(item)
      })} */}
      <StepBackLink setCurrent={handleStepBack}>{stepBackLink}</StepBackLink>
    </div>
  )
}

ResultsView.propTypes = {
  handleStepBack: PropTypes.func,
  ui: PropTypes.object,
  data: PropTypes.array,
}

export default ResultsView
