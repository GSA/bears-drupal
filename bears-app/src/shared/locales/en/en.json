{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start",
      "list": [
        {
          "notice": "<div>Your information<strong>will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "heading": "Your potential benefits",
    "description": "<div>You must verify your eligibility and visit each agency to apply. <strong>Check with your city, state, or employer for other benefits.</strong></div>",
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    }
  },
  "notEligibleResults": {
    "heading": "Benefits you did not qualify for",
    "description": "<div>According to your answers you <strong>are not</strong> eligible for these benefits.</div>",
    "heading2": "Non-eligible Results",
    "description2": "Check with your state, city, or employer for other benefits."
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  }
}
