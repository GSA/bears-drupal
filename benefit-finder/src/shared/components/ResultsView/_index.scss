@use '../../styles/colors/_index.scss' as color;
@use '../../styles/breakpoints/_index.scss' as *;
@use '../../styles/space/_index.scss' as space;

#benefit-finder {
  .result-view-details {
    padding: space.$space-lg;

    button.step-back-link {
      color: color.$medium-grey !important;
      background-color: color.$white !important;
      width: auto !important;
    }
  }

  .result-view-benefits,
  .result-view-unmet,
  .result-view-relvant-benefits,
  .result-view-share-results {
    padding: space.$space-md 0;
    border-bottom: 2px solid color.$base-light-grey;
  }

  .result-view-unmet {
    button {
      margin-bottom: space.$space-md;
    }

    p {
      padding-bottom: space.$padding-bottom-md;
    }
  }

  .result-view-share-results {
    text-align: center;
  }

  .result-view-share-results-button-group {
    button {
      margin-bottom: space.$space-sm;
    }
  }

  @media (width >= $desktop) {
    .result-view-details {
      padding: space.$space-md-plus 164px;
    }
  }
}
