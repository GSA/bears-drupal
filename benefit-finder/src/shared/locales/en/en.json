{
  "intro": {
    "heading": "Steps",
    "timeIndicator": "Time to complete",
    "steps": {
      "list": [
        {
          "title": "Answer questions"
        },
        {
          "title": "Review Results"
        },
        {
          "title": "Find out how to apply"
        }
      ]
    },
    "notices": {
      "heading": "Before you start:",
      "list": [
        {
          "notice": "<div>Your information<strong> will not</strong> be shared, saved, or submitted.</div>"
        },
        {
          "notice": "Provide complete and accurate information to get the most relevant results."
        },
        {
          "notice": "<div><strong>This form is not an application.</strong> You will need to apply for benefits with each agency.</div>"
        }
      ]
    },
    "button": "Start"
  },
  "stepIndicator": {
    "stepBackLink": "Back"
  },
  "requiredLabel": {
    "value": "Required"
  },
  "alertBanner": {
    "heading": "Error status",
    "description": "All fields marked required must be completed."
  },
  "buttonGroup": [
    {
      "value": "Back"
    },
    {
      "value": "Continue"
    }
  ],
  "reviewSelectionModal": {
    "heading": "Select an option:",
    "buttonGroup": [
      {
        "value": "Review selections"
      },
      {
        "value": "See results"
      }
    ]
  },
  "verifySelectionsView": {
    "heading": "Review and Confirm",
    "buttonGroup": [
      {
        "value": "Back"
      },
      {
        "value": "Confirm"
      }
    ]
  },
  "resultsView": {
    "qualified": {
      "chevron": {
        "heading": "Your Custom List",
        "description": "<div><p><strong>Here is your list of potential Federal Benefits.</strong> Your state, city or employer may offer other benefits. Visit each agency providing the benefit to learn if you are truly eligible and conduct an actual application.</p></div>"
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you meet the criteria to qualify for these benefits:</strong></div>"
    },
    "notQualified": {
      "chevron": {
        "heading": "Benefits You Did Not Qualify For",
        "description": ""
      },
      "heading": "Results",
      "description": "<div><strong>Based on your answers you did not meet the criteria to qualify for these benefits:</strong></div>"
    },
    "stepBackLink": "Back",
    "eligibleResults": {
      "heading": "Results",
      "description": "You may be eligible for these benefits."
    },
    "notEligibleResults": {
      "heading": "Benefits you did not qualify for",
      "description": "<div>Based on your answers you are not eligible for these benefits. <strong>You may become eligible if you enter additional information or your situation changes.</strong></div>",
      "cta": "See benefits you did not qualify for"
    },
    "relativeBenefits": {
      "heading": "Other benefits that might be relevant to you"
    },
    "shareResults": {
      "heading": "Sharing and printing",
      "shareButton": "Share link",
      "emailButton": "Email results",
      "emailSubject": "Results from USAGovâ€™s benefit finder",
      "printButton": "Print results",
      "description": "Copy a link to this page with the criteria you selected above. Make sure to share it only with those you trust as your answers will be visible."
    }
  },
  "shareResults": {
    "heading": "Share results",
    "description": "Your information will be viewable. Only share with those you trust.",
    "buttonGroup": [
      {
        "value": "Share link"
      },
      {
        "value": "Email results"
      }
    ]
  },
  "benefitAccordionGroup": {
    "closedState": "Open all",
    "openState": "Close all"
  },
  "sectionHeadings": {
    "start": "Complete Questions",
    "continue": "Questions Continued",
    "final": "Final Questions"
  },
  "select": {
    "labelSelect": "Select",
    "defaultValue": "Click to Select",
    "dateDefaultValue": "-Select-"
  },
  "date": {
    "alert": "Date should follow the format 01 - January 22 1977",
    "labelDay": "Day",
    "labelMonth": "Month",
    "labelYear": "Year",
    "monthOptions": [
      {
        "label": "01 - January"
      },
      {
        "label": "02 - February"
      },
      {
        "label": "03 - March"
      },
      {
        "label": "04 - April"
      },
      {
        "label": "05 - May"
      },
      {
        "label": "06 - June"
      },
      {
        "label": "07 - July"
      },
      {
        "label": "08 - August"
      },
      {
        "label": "09 - September"
      },
      {
        "label": "10 - October"
      },
      {
        "label": "11 - November"
      },
      {
        "label": "12 - December"
      }
    ]
  }
}
