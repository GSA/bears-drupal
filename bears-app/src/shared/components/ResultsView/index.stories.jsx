import ResultsView from './index.jsx'

export default {
  component: ResultsView,
  tags: ['autodocs'],
}

export const Primary = {}
