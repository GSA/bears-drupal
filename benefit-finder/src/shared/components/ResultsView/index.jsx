import { useEffect, useState } from 'react'
import * as apiCalls from '../../api/apiCalls'
import PropTypes from 'prop-types'
import {
  BenefitAccordionGroup,
  Button,
  EmailButton,
  Heading,
  StepBackLink,
  Card,
  Chevron,
  ShareButton,
} from '../index'
import createMarkup from '../../utils/createMarkup'
import './_index.scss'

/**
 * a functional component that renders a view of the form benefit state values
 * @component
 * @param {function} handlStepBack inherited ui translations
 * @param {object} ui inherited ui translations
 * @param {array} data inherited benefits data
 * @param {func} setBenefitsArray inherited state handler
 * @param {array} stepDataArray inherited state of inupt values from form entry
 * @return {html} returns a view page of filtered benefits
 */
const ResultsView = ({
  handleStepBack,
  ui,
  data,
  stepDataArray,
  relevantBenefits,
}) => {
  const {
    stepBackLink,
    notQualified,
    qualified,
    notEligibleResults,
    resultsRelativeBenefits,
    shareResults,
  } = ui

  const [notQualifiedView, setNotQualifiedView] = useState(false)

  const handleViewToggle = () => {
    setNotQualifiedView(!notQualifiedView)
    window.scrollTo(0, 0)
  }

  useEffect(() => {
    window.scrollTo(0, 0)
  }, [])

  // compare the selected criteria array with benefits
  return (
    <div className="result-view" data-testid="result-view">
      <Chevron
        heading={
          notQualifiedView === false
            ? qualified.chevron.heading
            : notQualified.chevron.heading
        }
        description={
          notQualifiedView === false
            ? qualified.chevron.description
            : notQualified.chevron.description
        }
      />
      <div className="grid-container">
        <div className="result-view-details">
          {notQualifiedView === false ? (
            <StepBackLink setCurrent={handleStepBack}>
              {stepBackLink}
            </StepBackLink>
          ) : (
            <Button
              className="step-back-link"
              onClick={() => setNotQualifiedView(false)}
              unstyled
            >
              {stepBackLink}
            </Button>
          )}
          <Heading className="result-view-heading" headingLevel={3}>
            {notQualifiedView ? notQualified.heading : qualified.heading}
          </Heading>
          <div
            className="result-view-description"
            dangerouslySetInnerHTML={
              notQualifiedView
                ? createMarkup(notQualified.description)
                : createMarkup(qualified.description)
            }
          />
          {/* map all the benefits into cards */}
          <div className="result-view-benefits">
            <BenefitAccordionGroup
              data={
                stepDataArray &&
                apiCalls.GET.ElegibilityByCriteria(
                  apiCalls.GET.SelectedValueAll(stepDataArray),
                  data
                )
              }
              entryKey={'benefit'}
              notQualifiedView={notQualifiedView}
              expandAll
              ui={ui}
            />
          </div>
          {notQualifiedView === false && (
            <div className="result-view-unmet">
              <Heading className="result-view-unmet-heading" headingLevel={3}>
                {notEligibleResults?.heading}
              </Heading>
              <p
                dangerouslySetInnerHTML={createMarkup(
                  notEligibleResults?.description
                )}
              />
              <Button onClick={handleViewToggle}>
                {notEligibleResults?.cta}
              </Button>
            </div>
          )}
          {relevantBenefits?.length > 0 && (
            <div className="result-view-relvant-benefits">
              <Heading
                className="result-view-relvant-benefits-heading"
                headingLevel={3}
              >
                {resultsRelativeBenefits?.heading}
              </Heading>
              {relevantBenefits && (
                <ul className="add-list-reset">
                  {relevantBenefits[0] && (
                    <div key="benefit-card-one">
                      <Card
                        className="relative-benefit-card"
                        title={`${relevantBenefits[0].lifeEvent.title}`}
                        body={`${relevantBenefits[0].lifeEvent.body}`}
                        cta={`${relevantBenefits[0].lifeEvent.cta}`}
                        href={`${relevantBenefits[0].lifeEvent.link}`}
                        carrotType={2}
                      ></Card>
                    </div>
                  )}
                  {relevantBenefits[1] && (
                    <div key="benefit-card-two">
                      <Card
                        className="relative-benefit-card"
                        title={`${relevantBenefits[1].lifeEvent.title}`}
                        body={`${relevantBenefits[1].lifeEvent.body}`}
                        cta={`${relevantBenefits[1].lifeEvent.cta}`}
                        href={`${relevantBenefits[1].lifeEvent.link}`}
                        carrotType={2}
                      ></Card>
                    </div>
                  )}
                </ul>
              )}
            </div>
          )}
          <div className="result-view-share-results">
            <Heading
              className="result-view-share-results-heading"
              headingLevel={3}
            >
              {shareResults?.heading}
            </Heading>
            <p>{shareResults?.description}</p>
            <div className="result-view-share-results-button-group">
              <ShareButton
                ui={shareResults}
                data={
                  stepDataArray && apiCalls.GET.SelectedValueAll(stepDataArray)
                }
              />
              <EmailButton
                ui={shareResults}
                data={
                  stepDataArray && apiCalls.GET.SelectedValueAll(stepDataArray)
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  )
}

ResultsView.propTypes = {
  handleStepBack: PropTypes.func,
  ui: PropTypes.object,
  data: PropTypes.array,
}

export default ResultsView
