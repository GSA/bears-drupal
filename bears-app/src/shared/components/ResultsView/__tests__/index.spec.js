import renderer from 'react-test-renderer'
import ResultsView from '../index.jsx'
import * as en from '../../../locales/en/en.json'

describe('ResultsView', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<ResultsView ui={en.resultsView} />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
