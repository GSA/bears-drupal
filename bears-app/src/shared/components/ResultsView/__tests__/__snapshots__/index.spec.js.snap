// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsView renders a match to the previous snapshot 1`] = `
<div>
  <h2
    className=""
  >
    Your potential benefits
  </h2>
  <div
    className="benefit-accordion-group"
  >
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      Expand all +
    </button>
  </div>
  <button
    className=""
    onClick={[Function]}
    type="button"
  >
    Back
  </button>
</div>
`;
